DeviceAdministration/Infrastructure.UnitTests/Infrastructure/IoTHubRepositoryTests.cs

```diff
@@ -0,0 +1,134 @@
+using Moq;
+using Ploeh.AutoFixture;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.Infrastructure
+{
+    public class IoTHubRepositoryTests
+    {
+        private readonly IIotHubRepository iotHubRepository;
+        private readonly Mock<IDeviceManager> deviceManagerMock;
+        private readonly Fixture fixture;
+
+        public IoTHubRepositoryTests()
+        {
+            this.deviceManagerMock = new Mock<IDeviceManager>();
+            this.deviceManagerMock.Setup(dm => dm.AddDeviceAsync(It.IsAny<Device>())).Returns(Task.FromResult(new Device()));
+            this.deviceManagerMock.Setup(dm => dm.RemoveDeviceAsync(It.IsAny<string>())).Returns(Task.FromResult(true));
+            this.iotHubRepository = new IotHubRepository(this.deviceManagerMock.Object);
+            this.fixture = new Fixture();
+        }
+
+        [Fact]
+        public async void AddDeviceAsync()
+        {
+            var device = this.fixture.Create<DeviceModel>();
+            var keys = new SecurityKeys("fbsIV6w7gfVUyoRIQFSVgw ==", "1fLjiNCMZF37LmHnjZDyVQ ==");
+            var sameDevice = await this.iotHubRepository.AddDeviceAsync(device, keys);
+            Assert.Equal(sameDevice, device);
+        }
+
+        [Fact]
+        public async void TryAddDeviceAsync()
+        {
+            var device = new Device("deviceId")
+                         {
+                             Authentication = null,
+                             Status = new DeviceStatus()
+                         };
+
+            var result = await this.iotHubRepository.TryAddDeviceAsync(device);
+            Assert.True(result);
+
+            result = await this.iotHubRepository.TryAddDeviceAsync(null);
+            Assert.False(result);
+        }
+
+        [Fact]
+        public async void GetIotHubDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            this.deviceManagerMock.Setup(dm => dm.GetDeviceAsync(deviceId))
+                .Returns(Task.FromResult(new Device(deviceId)));
+            var d = await this.iotHubRepository.GetIotHubDeviceAsync(deviceId);
+            Assert.Equal(deviceId, d.Id);
+        }
+
+        [Fact]
+        public async void RemoveDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            await this.iotHubRepository.RemoveDeviceAsync(deviceId);
+        }
+
+        [Fact]
+        public async void TryRemoveDeviceAsync()
+        {
+            var deviceId = this.fixture.Create<string>();
+            var result = await this.iotHubRepository.TryRemoveDeviceAsync(deviceId);
+            Assert.True(result);
+
+            this.deviceManagerMock.Setup(dm => dm.RemoveDeviceAsync(It.IsAny<string>())).Throws(new Exception());
+            result = await this.iotHubRepository.TryRemoveDeviceAsync(null);
```

Comment:
does TryRemoveDeviceAsync return false or throw an exception ? if latter, catch the expection.